### PR TITLE
Lowdb test fix

### PR DESCRIPTION
**Pull request by aj-r**
This fixes the Travis build errors that have been happening in all builds lately. Unfortunately this requires us to duplicate the `LoDashExplicitWrapper` type in order to support two different `write()` methods (one that returns a promise, and one that returns a value). These duplicated types are auto-generated so they can be easily updated when lodash changes. If anyone has a better solution, I'm all ears.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - Documentation for lodash changes: https://lodash.com/docs/4.17.5
  - Lowdb changes were just bug fixes; there should be no functional changes. Here's the documentation anyway: https://github.com/typicode/lowdb, https://github.com/typicode/lowdb/blob/master/examples/fp.md
- [ ] Increase the version number in the header if appropriate. N/A
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by aj-r** [comment]
It looks like I've accidentally included the changes from #25001 in this PR. I could try to make a new PR that excludes those commits, but typescript-bot seems to have closed that PR on me, and I'd still like to see those changes merged.

File: types/lowdb/lowdb-tests.ts [hunk 1]
```diff
@@ -124,37 +118,32 @@ const weDidNotBreakLodash: ExampleSchema["posts"] = lodashChain
...
+    sortBy<Post>("views") as PostsAction,
     take(5) as PostsAction
   ]);
 
-  const filtered: Post[] = dbFP("posts")(filter({ published: true }));
-  const writeAction: Post[] = dbFP("posts").write(concat({ id: "123" }));
-  const writeAction2: string = dbFP(["user", "name"]).write(set("typicode"));
+  const filtered: Post[] = dbFP("posts")(filter<Post>({ published: true }));
+  const writeAction: Post[] = dbFP("posts").write(concat<Post>({ id: 123 }));
+  const writeAction2: string = dbFP(["user", "name"]).write(() => "typicode");
```
Comment (aj-r) [inline types/lowdb/lowdb-tests.ts]: The documentation says to use `.write(set("typicode"))` here, but I think the documentation is wrong. Typescript generates an error if you do this, and rightly so in my opinion - doing this results in no value being written. `set()` is called internally by the `write()` function, so it shuldn't be needed.

File: types/lowdb/lowdb-tests.ts [hunk 2]
```diff
@@ -124,37 +118,32 @@ const weDidNotBreakLodash: ExampleSchema["posts"] = lodashChain
...
-  const postsFP: low.FpReturn<Post[], low.AdapterSync<DbSchema>> = dbFP(
-    "posts"
-  );
+  const postsFP = dbFP("posts");
 
   // replace posts with a new array resulting from concat
   // and persist database
   const write: Post[] = postsFP.write(
-    concat({ title: "lowdb is awesome", views: random(0, 5) })
+    concat<Post>({ title: "lowdb is awesome", views: random(0, 5) })
```
Comment (niieani) [inline types/lowdb/lowdb-tests.ts]: Is `<Post>` not inferred automatically, here and in other places?

File: types/lowdb/index.d.ts [hunk 3]
```diff
@@ -1,83 +1,129 @@
-// Type definitions for Lowdb 1.0.0
+// Type definitions for Lowdb 1.0
 // Project: https://github.com/typicode/lowdb
 // Definitions by: typicode <https://github.com/typicode>
 //                 Bazyli Brzóska <https://github.com/niieani>
```
Comment (niieani) [inline types/lowdb/index.d.ts]: You might want to add yourself @aj-r :)

**Review by niieani — APPROVED** [review]
Haven't tested it, but looks ok by eye. 👍

**Pull request merged**